src/main/java/com/minecolonies/coremod/network/messages/server/BuildToolPasteMessage.java

```diff
@@ -173,8 +180,21 @@ public void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogicalS
             if (isHut)
             {
                 handleHut(CompatibilityUtils.getWorldFromEntity(player), player, sn, rotation, pos, mirror, state, complete);
-                InstantStructurePlacer.loadAndPlaceStructureWithRotation(player.world, structureName,
-                  pos, BlockPosUtil.getRotationFromRotations(rotation), mirror ? Mirror.FRONT_BACK : Mirror.NONE, complete);
+                final Blueprint blueprint = MCCreativeStructureHandler.loadAndPlaceStructureWithRotation(player.world, structureName,
+                  pos, BlockPosUtil.getRotationFromRotations(rotation), mirror ? Mirror.FRONT_BACK : Mirror.NONE, !complete, player);
+
+                final TileEntity tileEntity = player.world.getTileEntity(pos);
```

Comment:
is this really needed? does the place not normally load the placed's TE data?

Reply:
Not of the one at the offset pos.

Reply:
offset pos? you mean the anchor position? why is it excluded? :D

Reply:
because the block gets placed first, to be placed with the right level data

Reply:
ye but upon that placement it should read the schematics TE data or not?

Reply:
No, that's why I've added this here.